agent/engine/docker_task_engine.go

```diff
@@ -625,7 +654,19 @@ func (engine *DockerTaskEngine) startContainer(task *api.Task, container *api.Co
 			Error: CannotStartContainerError{fmt.Errorf("Container not recorded as created")},
 		}
 	}
-	return client.StartContainer(dockerContainer.DockerID, startContainerTimeout)
+	dockerContainerMD := client.StartContainer(dockerContainer.DockerID, startContainerTimeout)
+
+	// Get metadata through container inspection and available task information then write this to the metadata file
+	// Performs this in the background to avoid delaying container start
+	if dockerContainerMD.Error == nil && engine.cfg.ContainerMetadataEnabled == true {
+		err := engine.metadataManager.UpdateMetadata(dockerContainer.DockerID, task, container)
+		if err != nil {
+			seelog.Errorf("Failed to update metadata file for container %s of task %s: %v", container, task, err)
```

Comment:
As with the reasoning above, this decision should depend on our defaults. However, unlike the above, since we currently fill the metadata file with basic information as a safety against update fails, it is arguable that we can continue the container launch here regardless of the metadata file's state.  Still, this goes against our above argument that an incomplete metadata file would hinder a user who deliberate enables this feature.